application/src/main/java/org/thingsboard/server/service/security/auth/LoggedInUsersLimitService.java

```diff
@@ -0,0 +1,78 @@
+/**
+ * Copyright © 2016-2021 The Thingsboard Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.thingsboard.server.service.security.auth;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.cache.Cache;
+import org.springframework.cache.CacheManager;
+import org.springframework.stereotype.Service;
+import org.thingsboard.server.common.data.CacheConstants;
+import org.thingsboard.server.common.data.id.UserId;
+
+import javax.annotation.PostConstruct;
+import java.util.Optional;
+
+@Service
+public class LoggedInUsersLimitService {
+    private final CacheManager cacheManager;
+    private Cache loginLimitCache;
+
+    private final Long maxLoginUsers;
+
+    public LoggedInUsersLimitService(CacheManager cacheManager,
+                                     @Value("${security.max_logged_in_users}") Long maxLoginUsers) {
+        this.cacheManager = cacheManager;
+        this.maxLoginUsers = maxLoginUsers;
+    }
+
+
+    @PostConstruct
+    protected void initCache() {
+        loginLimitCache = cacheManager.getCache(CacheConstants.LOGIN_AMOUNT_LIMIT_CACHE);
+    }
+
+    public boolean isOverLimit(UserId userId) {
+        if (maxLoginUsers == 0)
+            return false;
+        return getCurrentAmount(userId).map(currentAmount -> currentAmount >= maxLoginUsers).orElse(false);
+    }
+
+    public void decreaseCurrentLoggedInUsers(UserId userId) {
+        getCurrentAmount(userId).ifPresent(currentAmount -> {
+            loginLimitCache.put(toKey(userId), --currentAmount);
+            if (currentAmount == 0)
+                loginLimitCache.evict(toKey(userId));
+        });
+    }
+
+    public void increaseCurrentLoggedInUsers(UserId userId) {
+        Optional<Long> currentAmount = getCurrentAmount(userId);
+        String key = toKey(userId);
+        if (currentAmount.isPresent())
+            loginLimitCache.put(key, currentAmount.get() + 1);
+        else
+            loginLimitCache.put(key, 1L);
+    }
+
+    public Optional<Long> getCurrentAmount(UserId userId) {
+        return Optional.ofNullable(loginLimitCache.get(toKey(userId), Long.class));
+    }
```

Comment:
"Logged in users" does not reflect what this feature actually does. It is rather an amount of active sessions per one user. Please refactor wherever possible